### ISSUE TITLE
Panic error on the 6.2.X and 6.3.X branch, Assert error in CNT_Embark()

### ISSUE THREAD

**Issue by madrzejewski** [varnishcache/varnish-cache#3306]
Hello,

I'm opening this issue because I have random panic error from the child process. Varnish will start correctly and manage a few HTTP requests and then randomly crash. 

I was not able to reproduce this bug on my lab so I just have the error from the production server (without the debug info unfortunately). I was able to retrieve it from the journalctl, I hope it's complete. 

Similar issue : https://www.gitmemory.com/issue/varnishcache/varnish-cache/3191/575031257
Versions tried : 6.3.2 and 6.2.3

(more info after the error)

## Error 

```
Panic at: Fri, 01 May 2020 15:02:03 GMT
Assert error in CNT_Embark(), cache/cache_req_fsm.c line 1030:
  Condition((req->vcl0) == 0) not true.
version = varnish-6.3.2 revision 6870fd661a2b42c2e8adad838b5d92a71f27dccd, vrt api = 10.0
ident = Linux,3.10.0-1127.el7.x86_64,x86_64,-junix,-sfile,-sdefault,-hcritbit,epoll
now = 19769.033077 (mono), 1588345322.962847 (real)
Backtrace:
  0x43d8df: /usr/sbin/varnishd() [0x43d8df]
  0x4a3582: /usr/sbin/varnishd(VAS_Fail+0x42) [0x4a3582]
  0x4416c5: /usr/sbin/varnishd() [0x4416c5]
  0x468431: /usr/sbin/varnishd() [0x468431]
  0x45dfd3: /usr/sbin/varnishd() [0x45dfd3]
  0x45e490: /usr/sbin/varnishd() [0x45e490]
  0x7f19537e1ea5: /lib64/libpthread.so.0(+0x7ea5) [0x7f19537e1ea5]
  0x7f195350a8dd: /lib64/libc.so.6(clone+0x6d) [0x7f195350a8dd]
thread = (cache-worker)
pthread.attr = {
  guard = 4096,
  stack_bottom = 0x7efecf630000,
  stack_top = 0x7efecf63e000,
  stack_size = 57344,
}
thr.req = 0x7f194609b020 {
  vxid = 4489221, transport = HTTP/1 {
    state = HTTP1::Proc
  }
  step = R_STP_LOOKUP,
  req_body = R_BODY_NONE,
  restarts = 0, esi_level = 0,
  sp = 0x7f0029c0e820 {
    fd = 46, vxid = 4489220,
    t_open = 1588345321.520819,
    t_idle = 1588345321.520819,
    ws = 0x7f0029c0e860 {
      id = \"ses\",
      {s, f, r, e} = {0x7f0029c0e8a0, +96, (nil), +344},
    },
    transport = HTTP/1 {
      state = HTTP1::Proc
    }
    client = 109.234.xxx.xx 55751 :6081,
  },
  worker = 0x7efecf63cbd0 {
    ws = 0x7efecf63cc78 {
      id = \"wrk\",
      {s, f, r, e} = {0x7efecf63c380, +0, (nil), +2040},
    },
    VCL::method = DELIVER,
    VCL::return = deliver,
    VCL::methods = {},
  },
  ws = 0x7f194609b170 {
    id = \"req\",
    {s, f, r, e} = {0x7f194609d0b0, +1176, +57160, +57160},
  },
  http_conn = 0x7f194609d050 {
    fd = 46 (@0x7f0029c0e844),
    doclose = NULL,
    ws = 0x7f194609b170 {
      [Already dumped, see above]
    },
    {rxbuf_b, rxbuf_e} = {0x7f194609d0b0, 0x7f194609d35c},
    {pipeline_b, pipeline_e} = {(nil), (nil)},
    content_length = -1,
    body_status = none,
    first_byte_timeout = 0.000000,
    between_bytes_timeout = 0.000000,
  },
  http[req] = 0x7f194609b218 {
    ws = 0x7f194609b170 {
      [Already dumped, see above]
    },
    hdrs {
      \"GET\",
      \"/le-diabete/tout-savoir-sur-le-diabete/le-traitement/\",
      \"HTTP/1.1\",
      \"X-Retry: 1\",
      \"Host: www.domain.tld\",
      \"X-Real-IP: 66.249.xx.xx\",
      \"X-Custom-IP: yes\",
      \"X-SSL: yes\",
      \"Accept-Language: fr-BI\",
      \"Sec-Fetch-Dest: document\",
      \"Upgrade-Insecure-Requests: 1\",
      \"Purpose: prefetch\",
      \"Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9\",
      \"User-Agent: Mozilla/5.0 (Linux; Android 4.2.1; en-us; Nexus 5 Build/JOP40D) AppleWebKit/535.19 (KHTML, like Gecko; googleweblight) Chrome/38.0.1025.166 Mobile Safari/535.19\",
      \"X-Forwarded-Proto: https\",
      \"https: on\",
      \"X-Forwarded-For: 66.249.xx.xx, 109.234.xxx.xx\",
      \"X-Forwarded-Port: 80\",
      \"Accept-Encoding: gzip\",
    },
  },
  vdc = 0x7f194609d028 {
    nxt = (nil),
    retval = 0,
  },
  vcl = {
    name = \"vcl-domain-tld\",
    busy = 4,
    discard = 0,
    state = auto,
    temp = warm,
    conf = {
      syntax = \"40\",
      srcname = {
        \"/etc/varnish/vhosts/domain.tld.vcl\",
        \"Builtin\",
        \"/etc/varnish/acl.vcl\",
        \"/etc/varnish/routines.vcl\",
      },
    },
  },
  vcl0 = {
    name = \"vcl-root-010520\",
    busy = 30,
    discard = 0,
    state = auto,
    temp = warm,
    conf = {
      syntax = \"40\",
      srcname = {
        \"/etc/varnish/root.vcl\",
        \"Builtin\",
        \"/etc/varnish/acl.vcl\",
      },
    },
  },
  flags = {
  },
  privs = 0x7f194609b200 {
  },
  topreq = {
    req = 0x7f194609b020 {
      [Already dumped, see above]
    },
  },
},
thr.busyobj = (nil) {
},
vmods = {
  std = {0x7f1952cf4470, Varnish 6.3.2 6870fd661a2b42c2e8adad838b5d92a71f27dccd, 0.0},
},
```

## Context 

My configuration is the same as the one explained here : https://info.varnish-software.com/blog/one-vcl-per-domain
But, I have a lot of 'vhost' and different VCL file : more than 2000. So my option are not usual (max_vcl, cli_limit) : 

```
-p cli_limit=99999990 -p max_vcl=10000 -p thread_pool_min=5 -p thread_pool_max=500 -p thread_pool_timeout=300
```

The issue is possibly related to the different VCL (I have one per domain), or label (I have a lot of labels too), or the large number of VCLs loaded.

I will try to give you more info and a panic message with the debug symbol, but I have to change a few thing first on my side because it's a production server. 

OS: Centos7 - 3.10.0-1127.el7.x86_64
The package is installed from packagecloud.io

Thank you

**Comment by nigoroll** [comment]
Duplicate of #3019

**Comment by madrzejewski** [comment]
I agree but I'm using the varnish-6.3.2 version so it should already by fixed on this version no ? I also tested on 6.2.3 and if I understand correctly, it shoud also be fixed on this one too, but I'm still getting an error.

**Comment by nigoroll** [comment]
#3019 was fixed in 6.4.0

**Comment by madrzejewski** [comment]
Thanks. Sorry for the inconvenience.
